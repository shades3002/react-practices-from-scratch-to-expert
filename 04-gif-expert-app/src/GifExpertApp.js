import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => { 
    //     // setCategories([...categories, 'Batman']);
    //     setCategories( _categories => [...categories, 'Batman']);
    // }

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category => 
                        <GifGrid 
                            key={ category } 
                            category={ category } 
                        />
                    )
                }
            </ol>
        </Fragment>
    );
}

