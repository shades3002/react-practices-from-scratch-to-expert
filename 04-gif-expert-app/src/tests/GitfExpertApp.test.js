import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Test <GifExpertApp />', () => {
    
    test('should show correctly', () => {
        
        const wrapper = shallow(<GifExpertApp />) ;
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should show categories list', () => {
        
        const categories = ['One Punch', 'Drango Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />) ;

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })

});
