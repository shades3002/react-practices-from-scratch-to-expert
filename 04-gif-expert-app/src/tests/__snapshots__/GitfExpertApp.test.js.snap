// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> should show categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Drango Ball"
      key="Drango Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test <GifExpertApp /> should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
